gemp-swccg-cards/src/main/java/com/gempukku/swccgo/cards/set2/dark/Card2_101.java

```diff
@@ -0,0 +1,69 @@
+package com.gempukku.swccgo.cards.set2.dark;
+
+import com.gempukku.swccgo.cards.AbstractDroid;
+import com.gempukku.swccgo.cards.GameConditions;
+import com.gempukku.swccgo.cards.conditions.AboardCondition;
+import com.gempukku.swccgo.cards.effects.usage.OncePerPhaseEffect;
+import com.gempukku.swccgo.common.*;
+import com.gempukku.swccgo.filters.Filter;
+import com.gempukku.swccgo.filters.Filters;
+import com.gempukku.swccgo.game.PhysicalCard;
+import com.gempukku.swccgo.game.SwccgGame;
+import com.gempukku.swccgo.logic.actions.PlayInterruptAction;
+import com.gempukku.swccgo.logic.actions.TopLevelGameTextAction;
+import com.gempukku.swccgo.logic.conditions.Condition;
+import com.gempukku.swccgo.logic.effects.DeionizeStarshipEffect;
+import com.gempukku.swccgo.logic.modifiers.*;
+
+import java.util.Collections;
+import java.util.LinkedList;
+import java.util.List;
+
+/**
+ * Set: A New Hope
+ * Type: Character
+ * Subtype: Droid
+ * Title: R5-A2 (Arfive-Aytoo)
+ */
+public class Card2_101 extends AbstractDroid {
+    public Card2_101() {
+        super(Side.DARK, 4, 2, 1, 3, "R5-A2 (Arfive-Aytoo)");
+        setLore("Seldom used for navigation purposes, R5 units are known for their ability to perform hull maintenance and repair damage caused by ionization.");
+        setGameText("While aboard any starship, adds 1 to power and maneuver. During your control phase, if aboard your starship damaged by an Ion cannon, restores armor/maneuver and hyperspeed.");
+        addModelType(ModelType.ASTROMECH);
+        addIcons(Icon.A_NEW_HOPE);
+    }
+
+    @Override
+    protected List<Modifier> getGameTextWhileActiveInPlayModifiers(SwccgGame game, final PhysicalCard self) {
+        Condition aboardStarship = new AboardCondition(self, Filters.starship);
+        Filter starshipAboard = Filters.and(Filters.starship, Filters.hasAboard(self));
+
+        List<Modifier> modifiers = new LinkedList<Modifier>();
+        modifiers.add(new PowerModifier(self, starshipAboard, aboardStarship, 1));
+        modifiers.add(new ManeuverModifier(self, starshipAboard, aboardStarship, 1));
+        return modifiers;
+    }
+
+    @Override
+    protected List<TopLevelGameTextAction> getGameTextTopLevelActions(final String playerId, SwccgGame game, final PhysicalCard self, int gameTextSourceCardId) {
```

Comment:
This should be an automatic action instead of an optional top level action (same on the other droid)

And `List<PlayInterruptAction> actions = new LinkedList<PlayInterruptAction>();` should be removed

Reply:
Yes, it should happen automatically. The player can't choose when it happens and can't choose to not use this ability.

Reply:
Any progress on this?

Reply:
Sorry, life has been a maelstrom. Will try and get this tested and committed this weekend.